Sources/Swift/Integrations/SessionReplay/SentryOnDemandReplay.swift

```diff
@@ -21,52 +21,50 @@ enum SentryOnDemandReplayError: Error {
     case cantReadVideoSize
 }
 
-@available(iOS 16.0, tvOS 16.0, *)
 @objcMembers
 class SentryOnDemandReplay: NSObject {
     private let _outputPath: String
-    private let _onDemandDispatchQueue: DispatchQueue
-    
-    private var _starttime = Date()
-    private var _frames = [SentryReplayFrame]()
     private var _currentPixelBuffer: SentryPixelBuffer?
+    private var _totalFrames = 0
     
+    var frames = [SentryReplayFrame]()
+    var workingQueue: DispatchQueue
     var videoWidth = 200
     var videoHeight = 434
-    
     var bitRate = 20_000
     var frameRate = 1
     var cacheMaxSize = UInt.max
+    var dateProvider = SentryCurrentDateProvider()
     
     init(outputPath: String) {
         self._outputPath = outputPath
-        _onDemandDispatchQueue = DispatchQueue(label: "io.sentry.sessionreplay.ondemand")
+        workingQueue = DispatchQueue(label: "io.sentry.sessionreplay.ondemand")
     }
     
-    func addFrame(image: UIImage) {
-        _onDemandDispatchQueue.async {
-            self.asyncAddFrame(image: image)
+    func addFrameAsync(image: UIImage) {
+        workingQueue.async {
+            self.addFrame(image: image)
         }
     }
     
-    private func asyncAddFrame(image: UIImage) {
+    private func addFrame(image: UIImage) {
         guard let data = resizeImage(image, maxWidth: 300)?.pngData() else { return }
         
-        let date = Date()
-        let interval = date.timeIntervalSince(_starttime)
-        let imagePath = (_outputPath as NSString).appendingPathComponent("\(interval).png")
+        let date = dateProvider.date()
+        let imagePath = (_outputPath as NSString).appendingPathComponent("\(_totalFrames).png")
         do {
             try data.write(to: URL(fileURLWithPath: imagePath))
         } catch {
             print("[SentryOnDemandReplay] Could not save replay frame. Error: \(error)")
             return
         }
-        _frames.append(SentryReplayFrame(imagePath: imagePath, time: date))
+        frames.append(SentryReplayFrame(imagePath: imagePath, time: date))
         
-        while _frames.count > cacheMaxSize {
-            let first = _frames.removeFirst()
+        while frames.count > cacheMaxSize {
+            let first = frames.removeFirst()
             try? FileManager.default.removeItem(at: URL(fileURLWithPath: first.imagePath))
```

Comment:
`m`: Does it make sense to do some error handling here?

Reply:
We could flood even more the console log.

Honestly I dont think it matter, I clean up function to delete the entire replay folder is coming soon.